TACOptimizations/ControlFlowGraph.cs

```diff
@@ -0,0 +1,44 @@
+using System.Collections.Generic;
+
+namespace SimpleLang
+{
+    public class ControlFlowGraph
+    {
+        private List<BasicBlock> _basicBlocks;
+
+        private List<List<int>> _children;
+
+        private List<List<int>> _parent;
+        
+        public ControlFlowGraph()
+        {
+            _basicBlocks = new List<BasicBlock>();
+        }
+        
+        public ControlFlowGraph(List<BasicBlock> basicBlocks)
+        {
+            _basicBlocks = basicBlocks;
+        }
+
+        public List<BasicBlock> GetCurrentBasicBlocks() => _basicBlocks;
+
+        public List<BasicBlock> GetChildrenBasicBlock(int index)
+        {
+            var result = new List<BasicBlock>();
+            for (int j = 0; j < _children[index].Count; j++)
+                if (_children[index][j] == 1)
```

Comment:
why 1?
how are _children/_parent arranged? are they just matrices of 0/1?

Reply:
https://en.wikipedia.org/wiki/Adjacency_matrix 
the guys who will build the graph said that for now let it be so.

Reply:
If it's simply an adjacency matrix why do you need two of them? Parents and children can be decoded in one matrix just fine.

Reply:
Also `bool` is enough here, no need for `int`.

Reply:
Refer me to those guys who said to use adj.matrix, please. S.S. Mihalkovich said to NOT store CFG as matrix but maybe I missed something? :)

Reply:
These guys promised to remake in any case to their needs, so this is a temporary.